### PR TITLE
Add `protobuf`

### PR DESCRIPTION
**Pull request by jtpio**
This should fix https://github.com/emscripten-forge/recipes/issues/184

### CODE REVIEW

**Comment by tylere** [comment]
Also fixes #455

**Review by DerThorsten — APPROVED** [review]

File: recipes/recipes_emscripten/protobuf/recipe.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+package:
+  name: protobuf
+  version: "{{ version }}"
+
+build:
+  number: 0
+
+source:
+  - url: https://files.pythonhosted.org/packages/e2/86/44a1e4990a81cb4248a2091a182bb76a6417fddcaff560ceb6b44eb05c55/protobuf-{{ version }}.tar.gz
+    sha256: 23452f2fdea754a8251d0fc88c0317735ae47217e0d27bf330a30eec2848811a
```
Comment (proppy) [inline recipes/recipes_emscripten/protobuf/recipe.yaml:13]: should we add the test from the original conda-forge recipe as a following up PR?
https://github.com/conda-forge/protobuf-feedstock/blob/main/recipe/meta.yaml#L44-L54

File: recipes/recipes_emscripten/protobuf/recipe.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  version: 4.22.3
+
+package:
+  name: protobuf
+  version: "{{ version }}"
+
+build:
+  number: 0
+
+source:
```
Comment (proppy) [inline recipes/recipes_emscripten/protobuf/recipe.yaml:11]: curious why we don't need any of the `build` requirements from https://github.com/conda-forge/protobuf-feedstock/blob/main/recipe/meta.yaml#L32-L33 ?

File: recipes/recipes_emscripten/protobuf/recipe.yaml (hunk 1 above)
Comment (proppy) [inline recipes/recipes_emscripten/protobuf/recipe.yaml:13]: should we add `about` with `license` as a follow up PR? https://github.com/conda-forge/protobuf-feedstock/blob/main/recipe/meta.yaml#L56-L67

**Comment by proppy** [comment]
Looking at the CI logs, I'm curious if anything got built at all?
```
2023-05-13T14:24:14.5757315Z Downloading source
2023-05-13T14:24:14.5757618Z 
2023-05-13T14:24:14.5760559Z Source cache directory is: /home/runner/micromamba-root/envs/ci-env/conda-bld/src_cache
2023-05-13T14:24:14.5762722Z INFO:conda_build.source:Source cache directory is: /home/runner/micromamba-root/envs/ci-env/conda-bld/src_cache
2023-05-13T14:24:14.5764131Z Downloading source to cache: protobuf-4.22.3_23452f2fde.tar.gz
2023-05-13T14:24:14.5765115Z INFO:conda_build.source:Downloading source to cache: protobuf-4.22.3_23452f2fde.tar.gz
2023-05-13T14:24:14.5765952Z Downloading https://files.pythonhosted.org/packages/e2/86/44a1e4990a81cb4248a2091a182bb76a6417fddcaff560ceb6b44eb05c55/protobuf-4.22.3.tar.gz
2023-05-13T14:24:14.5767218Z INFO:conda_build.source:Downloading https://files.pythonhosted.org/packages/e2/86/44a1e4990a81cb4248a2091a182bb76a6417fddcaff560ceb6b44eb05c55/protobuf-4.22.3.tar.gz
2023-05-13T14:24:14.7522710Z INFO:conda_build.source:Success
2023-05-13T14:24:14.7522967Z Success
2023-05-13T14:24:14.9200229Z Extracting download
2023-05-13T14:24:14.9200753Z 
2023-05-13T14:24:14.9201882Z Preparing environment for protobuf
2023-05-13T14:24:14.9202418Z 
2023-05-13T14:24:14.9210015Z 
2023-05-13T14:24:14.9210374Z Starting build for protobuf
2023-05-13T14:24:14.9210627Z 
2023-05-13T14:24:14.9228184Z source tree in: 
2023-05-13T14:24:14.9228974Z /home/runner/micromamba-root/envs/ci-env/conda-bld/protobuf-0_1683987853545/work
2023-05-13T14:24:14.9720104Z number of files: 0
2023-05-13T14:24:14.9720751Z Fixing permissions
2023-05-13T14:24:14.9721215Z 
2023-05-13T14:24:14.9721409Z Adding files for protobuf
2023-05-13T14:24:14.9721767Z 
2023-05-13T14:24:14.9727588Z ATTENTION: No files added in target protobuf
2023-05-13T14:24:14.9730408Z 
2023-05-13T14:24:14.9730415Z 
2023-05-13T14:24:15.2078709Z Copying package to conda-build croot.  No packages otherwise alongside yours will be available unless you specify -c local.  To avoid this warning, your package must reside in a channel structure with platform-subfolders.  See more info on what a valid channel is at https://conda.io/docs/user-guide/tasks/create-custom-channels.html
2023-05-13T14:24:15.2080487Z WARNING:boa.core.test:Copying package to conda-build croot.  No packages otherwise alongside yours will be available unless you specify -c local.  To avoid this warning, your package must reside in a channel structure with platform-subfolders.  See more info on what a valid channel is at https://conda.io/docs/user-guide/tasks/create-custom-channels.html
2023-05-13T14:24:15.2954270Z INFO :: Time taken to mark (prefix)
2023-05-13T14:24:15.2954855Z         0 replacements in 0 files was 0.00 seconds
2023-05-13T14:24:15.2956052Z TEST START: /home/runner/micromamba-root/envs/ci-env/conda-bld/emscripten-32/protobuf-4.22.3-h51e9701_0.tar.bz2
2023-05-13T14:24:15.2956656Z Nothing to test for: /home/runner/micromamba-root/envs/ci-env/conda-bld/emscripten-32/protobuf-4.22.3-h51e9701_0.tar.bz2
2023-05-13T14:24:15.2956906Z 
2023-05-13T14:24:15.2956930Z 
2023-05-13T14:24:15.2956935Z 
2023-05-13T14:24:15.2957230Z                   Output: protobuf 4.22.3 BN: 0                  
2023-05-13T14:24:15.2957480Z                             Variant:                             
2023-05-13T14:24:15.2958579Z                              Build:                              
2023-05-13T14:24:15.2958831Z                                                                  
2023-05-13T14:24:15.2959188Z              ╷                     ╷          ╷       ╷          
2023-05-13T14:24:15.2959755Z   Dependency │ Version requirement │ Selected │ Build │ Channel  
2023-05-13T14:24:15.2960153Z  ════════════╪═════════════════════╪══════════╪═══════╪═════════ 
2023-05-13T14:24:15.2960462Z              ╵                     ╵          ╵       ╵          
2023-05-13T14:24:15.3044075Z Test recipe: {'package': {'name': 'protobuf', 'version': '4.22.3'}, 'build': {'number': 0}, 'source': [{'url': 'https://files.pythonhosted.org/packages/e2/86/44a1e4990a81cb4248a2091a182bb76a6417fddcaff560ceb6b44eb05c55/protobuf-4.22.3.tar.gz', 'sha256': '23452f2fdea754a8251d0fc88c0317735ae47217e0d27bf330a30eec2848811a'}], 'steps': [{'package': {'name': 'protobuf', 'version': '4.22.3'}}], 'recipe_file': '/tmp/tmp1fhmipc4/recipes/recipes_per_platform/protobuf/recipe.yaml', 'output_names': {'protobuf'}} in work_dir: /home/runner/work/recipes/recipes
2023-05-13T14:24:15.4382145Z Post job cleanup.
```

**Comment by jtpio** [comment]
Thanks @proppy.

The recipe follows the one from Pyodide available here: https://github.com/pyodide/pyodide/blob/main/packages/protobuf/meta.yaml, which also quite minimal.

But yeah we can try adding tests in a separate PR for example to check imports, to make sure it's built correctly.

**Pull request merged**